src/overlays/actors/ovl_En_Knight/z_en_knight.c

```diff
@@ -205,237 +380,4015 @@ ActorInit En_Knight_InitVars = {
+    if (limbIndex == KNIGHT_LIMB_TORSO) {
+        Matrix_MultVec3f(&sBodyCollider0ModelOffset, &colliderPos);
+        EnKnight_UpdateLimbCollider(this, 0, &this->bodyCollider, &colliderPos);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Knight/func_809B4BFC.s")
+    if (limbIndex == KNIGHT_LIMB_PELVIS) {
+        Matrix_MultVec3f(&sBodyCollider1ModelOffset, &colliderPos);
+        EnKnight_UpdateLimbCollider(this, 1, &this->bodyCollider, &colliderPos);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Knight/func_809B4C58.s")
+    if (limbIndex == KNIGHT_LIMB_SHIELD && this->actionFunc != EnKnight_IgosSitting) {
+        Matrix_MultVec3f(&sShieldColliderModelOffset, &colliderPos);
+        EnKnight_UpdateLimbCollider(this, 0, &this->shieldCollider, &colliderPos);
+        Matrix_MultVec3f(&sShieldParticlesModelOffset, &this->shieldParticlesPos);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Knight/func_809B4E84.s")
+    if (limbIndex == KNIGHT_LIMB_SWORD) {
+        Matrix_MultVec3f(&sSwordColliderModelOffset, &colliderPos);
+        EnKnight_UpdateLimbCollider(this, 0, &this->swordCollider, &colliderPos);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Knight/func_809B4ED8.s")
+    if (limbIndex == KNIGHT_LIMB_RIGHT_FOOT) {
+        Matrix_MultZero(&this->feetPositions[0]);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Knight/func_809B4F90.s")
+    if (limbIndex == KNIGHT_LIMB_LEFT_FOOT) {
+        Matrix_MultZero(&this->feetPositions[1]);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Knight/func_809B5058.s")
+void EnKnight_PostLimbDrawHead(PlayState* play, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx) {
+    static Vec3f sFocusPosModelOffset = { 300.0f, 500.0f, 0.0f };
+    EnKnight* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Knight/func_809B51DC.s")
+    if (limbIndex == KNIGHT_LIMB_NECK) {
+        Matrix_MultVec3f(&sFocusPosModelOffset, &this->actor.focus.pos);
+        Math_Vec3f_Copy(&this->breathBasePos, &this->actor.focus.pos);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Knight/func_809B52E8.s")
+void EnKnight_TransformLimbDraw(PlayState* play, s32 limbIndex, Actor* thisx, Gfx** gfx) {
+    EnKnight* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Knight/func_809B5634.s")
+    if (this == sIgosInstance) {
+        if (limbIndex == IGOS_LIMB_SHIELD) {
+            Matrix_Scale(this->shieldScale, this->shieldScale, this->shieldScale, MTXMODE_APPLY);
+        } else if (limbIndex == IGOS_LIMB_SWORD) {
+            Matrix_Scale(this->swordScale, this->swordScale, this->swordScale, MTXMODE_APPLY);
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Knight/func_809B5698.s")
+Gfx* EnKnight_BuildEmptyDL(GraphicsContext* gfxCtx) {
+    Gfx* gfx = GRAPH_ALLOC(gfxCtx, 2 * sizeof(Gfx));
```

Comment:
```suggestion
    Gfx* gfx = GRAPH_ALLOC(gfxCtx, 1 * sizeof(Gfx));
```
Since GRAPH_ALLOC aligns the size, does this work and seems more accurate?